codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2HeadersDecoderTest.java

```diff
@@ -175,6 +174,79 @@ public void decodingConnectionRelatedHeadersMustFailValidation() throws Exceptio
         verifyValidationFails(decoder, encode(b(":method"), b("GET"), b("te"), b("compress")));
     }
 
+    @Test
+    public void decodingInvalidHeaderValueMustFailValidation() throws Exception {
+        final DefaultHttp2HeadersDecoder decoder = new DefaultHttp2HeadersDecoder(true, true);
+
+        for (int illegalFirstChar = 0; illegalFirstChar < 0x21; illegalFirstChar++) {
+            verifyValidationFails(decoder, encode(b(":method"), b("GET"),
+                    b("test_header"), new byte[]{ (byte) illegalFirstChar, (byte) 'a' }));
+        }
+        verifyValidationFails(decoder, encode(b(":method"), b("GET"), b("test_header"), b("\u007Fa")));
+
+        for (int illegalSecondChar = 0; illegalSecondChar < 0x21; illegalSecondChar++) {
```

Comment:
nit: we could also generate these and pass as `param`. This way it is easier to re-run just the failed one.